extensions/http-curl/tests/HttpGetIntegrationTest.cpp

```diff
@@ -63,78 +62,134 @@ class HttpResponder : public CivetHandler {
   }
 };
 
-int main(int argc, char **argv) {
-  using org::apache::nifi::minifi::utils::verifyLogLinePresenceInPollTime;
-  const cmd_args args = parse_cmdline_args(argc, argv);
+class RetryHttpGetResponder : public CivetHandler {
+ public:
+  bool handleGet(CivetServer* /*server*/, struct mg_connection *conn) override {
+    puts("handle get with retry");
+    mg_printf(conn, "HTTP/1.1 501 Not Implemented\r\nContent-Type: text/plain\r\nContent-Length: 0\r\nConnection: close\r\n\r\n");
+    return true;
+  }
+};
 
-  LogTestController::getInstance().setDebug<core::Processor>();
-  LogTestController::getInstance().setDebug<core::ProcessSession>();
-  LogTestController::getInstance().setDebug<utils::HTTPClient>();
-  LogTestController::getInstance().setDebug<minifi::controllers::SSLContextService>();
-  LogTestController::getInstance().setDebug<minifi::processors::InvokeHTTP>();
-  LogTestController::getInstance().setDebug<minifi::processors::LogAttribute>();
+class VerifyHTTPGet {
```

Comment:
there seems to be a similar class `VerifyRetryInvokeHTTP` in `VerifyInvokeHTTPTests.cpp`, how do the two relate?

Reply:
VerifyInvokeHTTPTests is similar in design, but it has verifications with callbacks for POST requests and only uses yaml configuration files with POST invoke operations.

Reply:
could we somehow reuse it (or any of its base classes `VeifyInvokeHTTP`, `HTTPIntegrationBase`) as base class of `VerifyHTTPGet`? they provide some of functionality implemented here (setting up the server, possibly the flow itself)

Reply:
I wanted to keep it separate as the `HTTPIntegrationBase` has way more complexity than we need, but I'll check if we could reuse it and reduce any duplications.

Reply:
I did some refactoring and reused the `VeifyInvokeHTTP` class for the `VerifyHTTPGet` in 13b46dc6e39687ec63bd545127bae8a6012c7327